src/box/box.cc

```diff
@@ -4532,3 +4568,98 @@ box_read_ffi_enable(void)
 	if (--box_read_ffi_disable_count == 0)
 		box_read_ffi_is_disabled = false;
 }
+
+/**
+ * Drop the constraint or foreign key. If fieldno is equal to UINT32_MAX then
+ * the tuple constraint or tuple foreign key is dropped, otherwise the field
+ * constraint or field foreign key is dropped.
+ */
+static int
+constraint_drop(uint32_t space_id, const char *name, uint32_t name_len,
+		uint32_t fieldno, enum tuple_constraint_type type)
+{
+	const struct space *space = space_by_id(space_id);
+	uint32_t total_count;
+	struct tuple_constraint_def *cdefs;
+	if (fieldno != UINT32_MAX) {
+		assert(fieldno < space->def->field_count);
+		total_count = space->def->fields[fieldno].constraint_count;
+		cdefs = space->def->fields[fieldno].constraint_def;
+	} else {
+		total_count = space->def->opts.constraint_count;
+		cdefs = space->def->opts.constraint_def;
+	}
+	bool is_exists = false;
+	uint32_t count = 0;
+	for (uint32_t i = 0; i < total_count; ++i) {
+		if (cdefs[i].type != type)
+			continue;
+		++count;
+		uint32_t len = MIN(cdefs[i].name_len, name_len);
+		if (memcmp(cdefs[i].name, name, len) == 0)
+			is_exists = true;
+	}
+	if (!is_exists) {
+		int errcode = type == CONSTR_FKEY ? ER_NO_SUCH_FOREIGN_KEY :
+			      ER_NO_SUCH_CONSTRAINT;
+		diag_set(ClientError, errcode, tt_cstr(name, name_len),
+			 space_name(space));
+		return -1;
+	}
+
+	char key[16];
+	char *key_end = key + mp_format(key, 16, "[%u]", space_id);
+	struct region *region = &fiber()->gc;
+	uint32_t used = region_used(region);
+	const char *type_str = tuple_constraint_type_strs[type];
+	const char *path = fieldno != UINT32_MAX ?
+			   tt_sprintf("format[%u].%s", fieldno + 1, type_str) :
+			   tt_sprintf("flags.%s", type_str);
+	if (count != 1) {
+		uint32_t size = strlen(path) + name_len + 2;
+		char *buf = (char *)region_alloc(region, size);
+		if (buf == NULL) {
+			diag_set(OutOfMemory, size, "region_alloc", "buf");
+			return -1;
+		}
+		snprintf(buf, size, "%s.%*s", path, name_len, name);
```

Comment:
Nit: It'd be nice to have an assertion here checking that we don't write beyond the buffer.

Reply:
I added `assert(strlen(path) < size)`. Not sure if this enough since it will work properly only if length of `path` is actually less than `size`. Should I replace it with something like the code below?
```Diff
diff --git a/src/box/sql.c b/src/box/sql.c
index f11576f1c..4d3bf4f53 100644
--- a/src/box/sql.c
+++ b/src/box/sql.c
@@ -1524,14 +1524,16 @@ sql_constraint_path(struct region *region, uint32_t space_id, const char *name)
                        }
                        assert(type_count > 0);
                        const char *type_str = tuple_constraint_type_strs[type];
+                       uint32_t len;
                        if (type_count == 1) {
-                               snprintf(path, size, "format[%u].%s", i + 1,
-                                        type_str);
+                               len = snprintf(path, size, "format[%u].%s",
+                                              i + 1, type_str);
                        } else {
-                               snprintf(path, size, "format[%u].%s.%s", i + 1,
-                                        type_str, name);
+                               len = snprintf(path, size, "format[%u].%s.%s",
+                                              i + 1, type_str, name);
                        }
-                       assert(strlen(path) < size);
+                       assert(len < size);
+                       (void)len;
                        return path;
                }
        }
```